File: 1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
    var visited = make(map[int]int)
    
    for i, v := range nums {
        
        j, ok := visited[target - v]
        
        if ok { 
            return []int{i, j} 
        }
        
        visited[v] = i
        
    }
    
    return []int{0,1}
}

File: 1-two-sum/1-two-sum.ts
function twoSum(nums: number[], target: number): number[] {

    const cache = new Map()

    for (let i = 0; i < nums.length; i++) {
  
        let m = target - nums[i]
        
        if (cache.has(m)){
            return  [i, cache.get(m)]
        }
        
        cache.set(nums[i], i)
    }
   
};
